src/formatDate.js

```diff
@@ -50,7 +50,33 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const partOfDate = date.split(`${fromFormat[3]}`);
+  const dateObj = {};
+  const result = [];
+
+  for (let i = 0; i < partOfDate.length; i++) {
+    dateObj[fromFormat[i]] = partOfDate[i];
+  }
+
+  if (dateObj.hasOwnProperty('YYYY')) {
+    dateObj.YY = dateObj.YYYY.slice(2);
+  }
+
+  if (dateObj.YY < 30) {
+    dateObj.YYYY = '20' + dateObj.YY;
+  } else {
+    dateObj.YYYY = '19' + dateObj.YY;
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    for (const part in dateObj) {
+      if (toFormat[i] === part) {
+        result.push(dateObj[part]);
+      }
+    }
+  }
+
+  return result.join(`${toFormat[3]}`);
```

Comment:
Avoid using template literals for a single variable. Use 'toFormat[3]' directly.

Reply:
done